nipype/interfaces/afni/preprocess.py

```diff
@@ -46,12 +47,172 @@ class CentralityInputSpec(AFNICommandInputSpec):
+    an EPI and an anatomical structural dataset, and applies the resulting
+    transformation to one or the other to bring them into alignment.
+
+    This script computes the transforms needed to align EPI and  
+    anatomical datasets using a cost function designed for this purpose. The  
+    script combines multiple transformations, thereby minimizing the amount of 
+    interpolation applied to the data.
+    
+    Basic Usage:
+      align_epi_anat.py -anat anat+orig -epi epi+orig -epi_base 5
+    
+    The user must provide EPI and anatomical datasets and specify the EPI
+    sub-brick to use as a base in the alignment.  
+
+    Internally, the script always aligns the anatomical to the EPI dataset,
+    and the resulting transformation is saved to a 1D file. 
+    As a user option, the inverse of this transformation may be applied to the 
+    EPI dataset in order to align it to the anatomical data instead.
+
+    This program generates several kinds of output in the form of datasets
+    and transformation matrices which can be applied to other datasets if
+    needed. Time-series volume registration, oblique data transformations and
+    Talairach (standard template) transformations will be combined as needed
+    and requested (with options to turn on and off each of the steps) in
+    order to create the aligned datasets.
+
+    For complete details, see the `align_epi_anat.py' Documentation.
+    <https://afni.nimh.nih.gov/pub/dist/doc/program_help/align_epi_anat.py.html>`_
+
+    Examples
+    ========
+    >>> from nipype.interfaces import afni
+    >>> al_ea = afni.AlignEpiAnatPy()
+    >>> al_ea.inputs.py27_path = "/opt/miniconda/envs/py27/bin/python"
+    >>> al_ea.inputs.anat = "structural.nii"
+    >>> al_ea.inputs.in_file = "functional.nii"
+    >>> al_ea.inputs.epi_base = '0'
+    >>> al_ea.inputs.epi_strip = '3dAutomask'
+    >>> al_ea.inputs.volreg = 'off'
+    >>> al_ea.inputs.tshift = 'off'
+    >>> al_ea.inputs.save_skullstrip = True
+    >>> al_ea.cmdline # doctest: +ALLOW_UNICODE
+    'echo "" &&  /opt/miniconda/envs/py27/bin/python /root/abin/align_epi_anat.py -anat structural.nii -epi_base 0 -epi_strip 3dAutomask -epi functional.nii -save_skullstrip -tshift off -volreg off'
+    >>> res = allineate.run()  # doctest: +SKIP
+    """
+    _cmd = 'echo "" && '
+    input_spec = AlignEpiAnatPyInputSpec
+    output_spec = AlignEpiAnatPyOutputSpec
+
+    def _list_outputs(self):
+        outputs = self.output_spec().get()
+        anat_prefix = ''.join(self._gen_fname(self.inputs.anat).split('+')[:-1])
+        epi_prefix = ''.join(self._gen_fname(self.inputs.in_file).split('+')[:-1])
+        ext = '.HEAD'
+        matext='.1D'
+        if not isdefined(self.inputs.suffix):
+            suffix = '_al'
+        else:
+            suffix = '_'+self.inputs.suffix
+        if self.inputs.anat2epi:
+            outputs['anat_al_orig'] = os.path.abspath(self._gen_fname(anat_prefix, suffix=suffix+'+orig')+ext)
```

Comment:
[AFNICommand._gen_fname](https://twitter.com/deleteuracct/status/890807118994878464) already prepends the cwd, so you can drop the `os.path.abspath`. I'm a little confused by the `ext` here. It looks like, if `outputtype == 'NIFTI', the filename will end with `_al+orig.nii.HEAD`. I'm not very familiar with AFNI's formats, but I'd assume `.HEAD` and `.nii` are mutually exclusive.

Perhaps what you want is:

```Python
outputtype = self.inputs.outputtype
ext = '.HEAD' if outputtype == 'AFNI' else Info.output_type_to_ext(outputtype)
[...]
outputs['anat_al_orig'] = self._gen_fname(anat_prefix, suffix=suffix + '+orig', ext=ext)
```

And so-on for the rest.